storage/innobase/handler/handler0alter.cc

```diff
@@ -5886,6 +5886,21 @@ ha_innobase::commit_inplace_alter_table(
 	row_mysql_unlock_data_dictionary(trx);
 	trx_free_for_mysql(trx);
 
+	/* Rebuild index translation table now for temporary tables if we are
+	restoring secondary keys, as ha_innobase::open will not be called for
+	the next access.  */
+	if (DICT_TF2_FLAG_IS_SET(ctx0->new_table, DICT_TF2_TEMPORARY)
+	    && ctx0->num_to_add_index) {
```

Comment:
Optional but 5.7 we are enforcing rule that only boolean should be used w/o RHS.
Should if possible to make the condition more clear mentioning it as

(ctx0->num_of_add_index > 0)

As this is not a official rule will leave decision to developer.